File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = prices[0];
        int maxProfit = 0;
        for(int i=1;i<prices.length;i++){
            int diff = prices[i]-minPrice;
            maxProfit = Math.max(maxProfit,diff);
            minPrice = Math.min(prices[i],minPrice);
        }
        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit = 0;
        int minPrice = prices[0];
        for(int i=0;i<prices.size();i++){
            int ans = prices[i] - minPrice;
            maxProfit = max(maxProfit,ans);
            minPrice = min(minPrice,prices[i]);
        }
        return maxProfit;
    }
};